code/Test_definitions/Geofencing.feature

```diff
@@ -0,0 +1,179 @@
+    Given the request body is not available and path parameter "subscriptionId" is set to the identifier of a non-existing subscription which is in invalid format
+    When the request "deleteGeofencingSubscription" is sent
+    Then the response  code is 400
+    And the response property "$.status" is 400
+    And the response property "$.code" is "INVALID_ARGUMENT"
+    And the response property "$.message" contains a user friendly text
+
+  @geofencing_subscriptions_13_no_authorization_header
+  Scenario: No Authorization header
+    Given a valid subscription request body and header "Authorization" is not available
+    When the  request "createSubscription" is sent 
+    Then the response status code is 401
+    And the response property "$.status" is 401
+    And the response property "$.code" is "UNAUTHENTICATED"
+    And the response property "$.message" contains a user friendly text
+
+
+  @geofencing_subscriptions_14_expired_access_token
+  Scenario: Expired access token
+    Given a valid subscription request body and header "Authorization" is expired
+    When the  request "createSubscription" is sent
+    Then the response status code is 401
+    And the response property "$.status" is 401
+    And the response property "$.code" is "UNAUTHENTICATED"
+    And the response property "$.message" contains a user friendly text
+
+
+  @geofencing_subscriptions_15_invalid_access_token
+  Scenario: Invalid access token
+    Given a valid subscription request body and header "Authorization" set to an invalid access token
+    When the  request "createSubscription" is sent
+    Then the response status code is 401
+    And the response header "Content-Type" is "application/json"
+    And the response property "$.status" is 401
+    And the response property "$.code" is "UNAUTHENTICATED"
+    And the response property "$.message" contains a user friendly text
+
+   @geofencing_subscriptions_16_sink_credentials
+  Scenario:   Validate that the subscribed events are received in the sink, with the right sinkCredential, for those situations specified in the API.
+    Given a valid subscription request body with "sink" and "sinkcredentials"  
+    When the  request "createSubscription" is sent  
+    Then the response  code is 201 and subscription is created
+    Then the  event details are observed on notifications-url
+    And the subscribed event  received on notifications-callback-url & sink credentials are as expected
+
+ @geofencing_subscriptions_17_subscriptionExpireTime
+  Scenario:   For subscriptions that provide subscriptionExpireTime, validate that the subscribed events are not longer received after the expiration time.
+    Given a valid subscription request body and for less $.duration
+    When the request "createSubscription" is sent
+    Then the subscription is expired
+    And no event notifications received on callback-url when device enters or leaves geofence
```

Comment:
For me we should replace your line 164 with:

```
 And The callback notification application receives subscription-ends event at provided callbackUrl
 And notification body complies with the OAS schema at "##/components/schemas/CloudEvent"
 And type="org.camaraproject.geofencing-subscriptions.v0.subscription-ends"
```

indeed in this case a subscription-ends has to be sent to the callback url.

Reply:
I would rephrase the scenario as:

Scenario: Events are not longer received for a created subscription after subscriptionExpireTime
Given an existing subscription to area-entered for a device and an area with explicit subscriptionExpireTime
When the device enters the area after that subscriptionExpireTime
Then an event is not received in the subscription callback url

Additionally we could add a new scenario that covers what Ludovic mentions, to check that the event for subscription-ends is received, something like:

Scenario: subscription-ends event is received for a created subscription at subscriptionExpireTime
Given an existing subscription with explicit subscriptionExpireTime
When subscriptionExpireTime is reached
Then an event is received in the subscription callback url
And the event request body complies with the schema at "/components/schemas/EventSubscriptionEnds"
And the event terminationReason is SUBSCRIPTION_EXPIRED